File: Data Structures & Algorithms/validate-parentheses/submission-6.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack=[]
        ss={"}":"{","]":"[",")":"("}
        for i in s:
            if i in ss.values():
                stack.append(i)
            elif i in ss:
                if not stack or ss[i]!=stack[-1]:
                    return False
                stack.pop()    
                
        return not stack

File: Data Structures & Algorithms/daily-temperatures/submission-6.java
class Solution {
    public int[] dailyTemperatures(int[] temperatures) {
        int n= temperatures.length;
        int[] res = new int[n];
        Stack<int[]>stack = new Stack<>();
        for(int i=0;i<n;i++){
            while (!stack.isEmpty() && stack.peek()[0] < temperatures[i]){
                    int[] pair = stack.pop();
                    res[pair[1]] = i- pair[1];
            }
            stack.push( new int[]{temperatures[i],i});
        }
        return res;
    }
}
